+nosnoc/+dcs/Gcs.m

```diff
@@ -0,0 +1,67 @@
+classdef Gcs < nosnoc.dcs.Base
+    properties
+        lambda
+
+        f_x
+
+        dims
+
+        f_fun
+        nabla_c_fun
+        c_fun
+        g_comp_path_fun
+    end
+
+    methods
+        function obj = Gcs(model)
+            obj.model = model;
+            obj.dims = model.dims;
+        end
+        
+        function generate_variables(obj, opts)
+            import casadi.*
+            dims = obj.dims;
+            % dimensions
+            dims.n_lambda = dims.n_c;
+            obj.lambda = define_casadi_symbolic(opts.casadi_symbolic_mode,'lambda',dims.n_lambda);
+
+            obj.dims = dims;
+        end
+
+        function generate_equations(obj, opts)
+            import casadi.*
+            model = obj.model;
+            dims = obj.dims;
+
+            nabla_c = model.c.jacobian(model.x)';
+            
+            obj.f_x = model.f_x + model.E*nabla_c*obj.lambda;
+
+            obj.f_x_fun = Function('f_x', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {obj.f_x, model.f_q});
+            obj.f_fun = Function('f', {model.x, model.z, model.u, model.v_global, model.p}, {model.f_x});
+            obj.f_q_fun = Function('f_q', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {model.f_q});
+            obj.g_z_fun = Function('g_z', {model.x, model.z, model.u, model.v_global, model.p}, {model.g_z});
+            obj.g_alg_fun = Function('g_alg', {model.x, model.z, obj.lambda, model.u, model.v_global, model.p}, {[]});
```

Comment:
why is this empty output function defined? (used for latter developments like sdf) 

also comments what is in g_z and what in g_alg stored would be of great help.

Reply:
Partially yes. In this case this is because in a GCS we don't have any non-complementarity components that are used to define the algebraics $\lambda$, hence the algorithmic algebraic function is empty.

Reply:
Then it makes sense to not have it at all, instead of keeping track of any empty think.
Except, algorithmic algebraic conditions appear in the foreseeable future.